src/KDSoapClient/KDSoapClientInterface.cpp

```diff
@@ -142,12 +144,28 @@ QNetworkRequest KDSoapClientInterfacePrivate::prepareRequest(const QString &meth
     return request;
 }
 
-QBuffer *KDSoapClientInterfacePrivate::prepareRequestBuffer(const QString &method, const KDSoapMessage &message, const KDSoapHeaders &headers)
+QBuffer *KDSoapClientInterfacePrivate::prepareRequestBuffer(const QString &method, const KDSoapMessage &message, const QString &soapAction, const KDSoapHeaders &headers)
 {
     KDSoapMessageWriter msgWriter;
     msgWriter.setMessageNamespace(m_messageNamespace);
     msgWriter.setVersion(m_version);
-    const QByteArray data = msgWriter.messageToXml(message, (m_style == KDSoapClientInterface::RPCStyle) ? method : QString(), headers, m_persistentHeaders, m_authentication);
+    QByteArray data;
+    if (m_useWS_Addressing) {
+        KDSoapMessage msg = message;
+        KDSoapMessageAddressingProperties prop = message.messageAddressingProperties();
+        prop.setAction(soapAction);
+        msg.setMessageAddressingProperties(prop);
+
+        data = msgWriter.messageToXml(msg,
+                                                   (m_style == KDSoapClientInterface::RPCStyle) ? method : QString(),
+                                                   headers, m_persistentHeaders,
+                                                   m_authentication);
+    } else {
+        data = msgWriter.messageToXml(message,
+                                                   (m_style == KDSoapClientInterface::RPCStyle) ? method : QString(),
+                                                   headers, m_persistentHeaders,
+                                                   m_authentication);
```

Comment:
I think it is not necessary to duplicate this line.

Reply:
Fixed.